workers/caching/metaphysics-cdn.test.ts

```diff
@@ -0,0 +1,108 @@
+import { jest } from "@jest/globals"
+
+describe("metaphysics-cdn fetch handler", () => {
+  let fetchHandler: (request: Request, env: any, ctx: any) => Promise<Response>
+  let mockCache: any
+
+  beforeEach(async () => {
+    // Import the fetch handler
+    const module = await import("./metaphysics-cdn")
+    fetchHandler = module.default.fetch
+
+    // Mock the caches.default object
+    mockCache = {
+      match: jest.fn(),
+      put: jest.fn(),
+    }
+    global.caches = { default: mockCache }
+
+    // Mock the crypto.subtle.digest function
+    global.crypto = {
+      subtle: {
+        digest: jest.fn(async (algorithm, _data) => {
+          if (algorithm !== "SHA-256") {
+            throw new Error("Unsupported algorithm")
+          }
+          // Mock a hash result (32 bytes for SHA-256)
+          return new Uint8Array(32).fill(1).buffer
+        }),
+      },
+    }
```

Comment:
Out of curiosity, does this need to be mocked because it's only a browser API?

Reply:
Thank you for asking! Yes, it needs to be mocked. The `crypto.subtle.digest` is a browser- native API.